test/Password/BcryptBCTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace ZendTest\Crypt\Password;
+
+use Zend\Crypt\Password\Bcrypt;
+use Zend\Math\Rand;
+
+/**
+ * @group      Zend_Crypt
+ */
+class BcryptBCTest extends \PHPUnit_Framework_TestCase
+{
+    public function setUp()
+    {
+        $this->bcrypt = new Bcrypt();
+    }
+
+    public function testBackwardCompatibilityV2()
+    {
+        $hash = $this->bcryptV2Implementation('test', 10);
+        $this->assertTrue($this->bcrypt->verify('test', $hash));
+    }
+
+    /**
+     * This is the Bcrypt::create implementation of ZF 2.*
```

Comment:
Nice — this will go a long ways towards ensuring users have a smooth upgrade. :grin: